### PR TITLE
Change EffectComposer to use the renderer's canvas size

### PR DESCRIPTION
**Pull request by greggman**
EffectComposer was using window.innerWidth and window.innerHeight which is not portable
if your app is not using the full window.

Unfortunately the devicePixelRatio crap messes things up as usual :(
I really wish you guys would remove any reference to devicePixelRatio from three.js
all it does is make it harder to follow best practices because you have to add magic
compensation outside for magic happening under the hood :(

In other words because three.js is deciding on a width and height on its own magic
instead of width and height specified by the user the user then has to guess
on how to compute a width and height that matches what three.js actually made

### CODE REVIEW

**Comment by mrdoob** [comment]
We'll get there! We'll get there!
Thanks! ^^

**Comment by mrdoob** [comment]
Also, I would say that `EffectComposer` using `window.innerWidth` was more of an issue here ;)

**Comment by WestLangley** [comment]
Why are you dividing by `pixelRatio`?

    var width  = Math.floor( renderer.context.canvas.width  / pixelRatio ) || 1;
    var height = Math.floor( renderer.context.canvas.height / pixelRatio ) || 1;

**Comment by greggman** [comment]
Because otherwise the effect changes. The old way was window.innerWidth which is in CSS pixels. The new way is canvas.width which WebGLRenderer has multiplied by pixelRatio so I need to divide to get the same effect as before

**Comment by WestLangley** [comment]
> so I need to divide to get the same effect as before

I would not worry about duplicating what we had before, because "before" was likely not right.

For me, this PR causes the Effect composer's render target to be of a different size than the renderer's drawing buffer. I would have thought we would want them to be the same size.

That being said, this `pixelRatio` thing is such a mess. I agree with you: it should be removed from the library.

**Comment by greggman** [comment]
They weren't the same size before this PR.

Before, on a HD-DPI machine the code would call `renderer.setSize(window.innerWidth, window.innerHeight)` and internally WebGLRenderer would set the canvas's width and height to those * pixelRatio. Then EffectComposer would make a render target that is `window.innerWidth` and `window.innerHeight` but those ones were not multiplied by pixelRatio. 

With this PR you should get exactly the same results. If you're not I screwed up :( 

The difference is with this PR if you're not using the full window EffectComposer will still work. For example [the editor](http://threejs.org/editor/)
